src/libutil/config.hh

```diff
@@ -241,6 +241,11 @@ protected:
     const T defaultValue;
     const bool documentDefault;
 
+    /**
+     * Protected internal set method which does the actual `T`-specific work.
+     */
+    virtual void setInternal(const std::string & str, bool append = false);
```

Comment:
Hmm, this is a hard one to name. Now it sounds like it's making something internal, which isn't quite right.
How about `parse`? I think that's a reasonably accurate name. Less general than `set`, so that's good. It does set things, which is not great, but not terrible as it follows from the signature that it does.
Could we make `parse` return `T` and make `set` responsible for actually setting the `value`?

Reply:
Great idea! I made `parse` and `appendOrSet`.